### DISCUSSION TITLE
slots in RAC buttons

### DISCUSSION THREAD

**Discussion by stefanprobst** [adobe/react-spectrum#4824] [General]
the `react-spectrum` `<Button>` (and others, like `Badge`) has dedicated slots for `<Text>` and `<Icon>` children -- is this a pattern you plan to carry over to `react-aria-components` as well? i am asking because currently the RAC button does not seem to provide slots for child content. thanks!

**Comment by devongovett** [comment]
The slots in spectrum are for layout purposes - the icon and text have styles that automatically get associated with via that slot. For `react-aria-components` this is not really a thing because it is a style-less library. We only provide slots when they have semantic meaning. In the case of a button, the icon and text are purely visual so they don't really concern RAC. You could create your own slot-like system for this similar to spectrum, however. We just use React context for that.
